### PR TITLE
Issues #13 and #14, Querying google calendar events, Displaying them in a recycler view, Refactoring packages

### PR DESCRIPTION
**Pull request by makchamp**
Issues #13 and #14: 

- Created Calendar Models : Calendar and CalendarEvent
- CalendarRepository, queries calendar data (like events) using contentResolver
- A Calendar is uniquely identified by the color of its events (set on google calendar app)
- Right now all custom events are being 'synced' (meaning all events created by the user on their Google Calendar App are being retrieved by the query)
- Selecting different calendars will be possible by filtering the calendar events by their color
- Added a recycler view which holds calendar events (in cardviews)
- The CalendarAdapter handles the recycler view items
- The event title, location and date/time are shown
 The cards have the same border color as the event color in the google calendar app
- Clicking on a event card will navigate to the map fragment (and later start the navigation to the destination if it is a valid location)
- Split the classes into different packages, views, models, viewmodels, adapters and repositories
- Asking for calendar read/write permission once the calendar fragment is opened

### CODE REVIEW

**Comment by NicholasNagy** [comment]
I'm busy today, so @Amanda-Kolopa if you can review the PR today that would be great!

File: app/src/main/java/com/droidhats/campuscompass/MainActivity.kt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package com.droidhats.campuscompass
 
 import android.os.Bundle
+import android.widget.Toast
```
Comment (Amanda-Kolopa) [inline app/src/main/java/com/droidhats/campuscompass/MainActivity.kt]: Not sure why we're importing Toast widget. This should be removed.

File: app/src/main/java/com/droidhats/campuscompass/repositories/CalendarRepository.kt [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+   // Singleton instantiation
+        private var instance: CalendarRepository? = null
+
+        fun getInstance() =
+            instance
+                ?: synchronized(this) {
+                instance
+                    ?: CalendarRepository().also { instance = it }
+            }
+    }
```
Comment (Amanda-Kolopa) [inline app/src/main/java/com/droidhats/campuscompass/repositories/CalendarRepository.kt]: This seems to be implementing the thread-safe variant of singleton. 
If this is the case, a disadvantage is that it performs synchronization at every call to CalendarRepository.
We can improve this to make an instance of type enum instead (enum variant of singleton).

File: app/src/main/java/com/droidhats/campuscompass/viewmodels/CalendarViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    private val context = getApplication<Application>().applicationContext
+    private var userCalendars = MutableLiveData<MutableMap<String, Calendar>>()
+
+    private val _text = MutableLiveData<String>().apply {
+        value = "Calendar"
+    }
+    val text: LiveData<String> = _text
+
+   fun init()
+    {
```
Comment (Amanda-Kolopa) [inline app/src/main/java/com/droidhats/campuscompass/viewmodels/CalendarViewModel.kt]: We need to follow a consistent coding format.
Please keep the starting curly bracket on the same line.

File: app/src/main/java/com/droidhats/campuscompass/viewmodels/CalendarViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+       userCalendars = CalendarRepository.getInstance().getCalendars(context)
+    }
+
+    fun getUserCalendars() : MutableLiveData<MutableMap<String, Calendar>>
+    {
```
Comment (Amanda-Kolopa) [inline app/src/main/java/com/droidhats/campuscompass/viewmodels/CalendarViewModel.kt]: Same comment:
We need to follow a consistent coding format.
Please keep the starting curly bracket on the same line.

File: app/src/main/java/com/droidhats/campuscompass/views/CalendarFragment.kt [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+class CalendarFragment : Fragment() {
+
+    private lateinit var calendarViewModel: CalendarViewModel
+
+    private var columnCount = 1
+
+    companion object {
+        private const val READ_CALENDAR_PERMISSION_REQUEST_CODE = 1
+        // This callback could also be private and be set on the host(main) Activity
+        var onCalendarEventClickListener: OnCalendarEventClickListener? = null
```
Comment (Amanda-Kolopa) [inline app/src/main/java/com/droidhats/campuscompass/views/CalendarFragment.kt]: I think we should keep it in this file.
The onCalendarEventClickListener is specific to the CalendarFragment also having this and future EventClickListeners in the main overtime would be messy.

File: app/src/main/res/layout/calendar_recycler_item.xml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+    </LinearLayout>
+
+</androidx.cardview.widget.CardView>
+
+
+
+
+
+
```
Comment (Amanda-Kolopa) [inline app/src/main/res/layout/calendar_recycler_item.xml]: Please watch out for unnecessary empty lines like these.

**Pull request merged**